cache/sampleStorePersist.js

```diff
@@ -17,21 +17,38 @@ const Sample = require('../db').Sample;
 const redisClient = require('./redisCache').client.sampleStore;
 const samsto = require('./sampleStore');
 const constants = samsto.constants;
+const log = require('winston');
 
 /**
- * Checks whether the redis sample store is currently being persisted to the
- * db.
+ * Truncate the sample table in the DB and persist all the samples from redis
+ * into the empty table.
  *
- * @returns {Promise} resolves to true if the redis sample store is currently
- *  being persisted to the db.
+ * @returns {Promise} - which resolves to true once the sample is persisted to
+ * the database
  */
-function persistInProgress() {
-  return redisClient.getAsync(constants.persistInProgressKey);
-} // persistInProgress
+function storeSampleToDb() {
+  log.info('Persis to db started :|. This will start by truncating the ' +
+    'sample table followed by persisting the sample to db');
+  return Sample.destroy({ truncate: true, force: true })
+  .then(() => {
+    log.info('truncated the sample table :|');
+    return redisClient.smembersAsync(constants.indexKey.sample);
+  })
+  .then((keys) => keys.map((key) => ['hgetall', key]))
+  .then((cmds) => redisClient.batch(cmds).execAsync())
+  .then((res) => {
+    const samplesToCreate = res.map((sample) => {
+      sample.relatedLinks = JSON.parse(sample.relatedLinks);
+      return sample;
+    });
+    return Sample.bulkCreate(samplesToCreate);
+  })
```

Comment:
If fixing the data issue on samples does not fix the PHX.SP3 samples problem, we need to change this to Sample.bulkUpsertByName (but this takes about 5min to 10mins running locally). Considering that none of the hooks are called for bulkCreate, I am surprised that it even works for us.